pxtblocks/blocklyloader.ts

```diff
@@ -540,48 +512,117 @@ namespace pxt.blocks {
         }
     }
 
+    function setHelpResources(block: any, id: string, name: string, tooltip: any, url: string) {
+        if (tooltip) block.setTooltip(tooltip);
+        if (url) block.setHelpUrl(url);
+
+        let tb = document.getElementById('blocklyToolboxDefinition');
+        let xml: HTMLElement = tb ? tb.querySelector(`category block[type~='${id}']`) as HTMLElement : undefined;
+        block.codeCard = <pxt.CodeCard>{
+            header: name,
+            name: name,
+            software: 1,
+            description: goog.isFunction(tooltip) ? tooltip() : tooltip,
+            blocksXml: xml ? (`<xml xmlns="http://www.w3.org/1999/xhtml">` + (cleanOuterHTML(xml) || `<block type="${id}"</block>`) + "</xml>") : undefined,
+            url: url
+        };
+    }
+
+    function installHelpResources(id: string, name: string, tooltip: any, url: string) {
+        let block = Blockly.Blocks[id];
+        let old = block.init;
+        if (!old) return;
+
+        block.init = function () {
+            old.call(this);
+            let block = this;
+            setHelpResources(this, id, name, goog.isFunction(tooltip) ? function() { return tooltip(block); } : tooltip, url);
+        }
+    }
+
     function initLoops() {
+        let msg:any = Blockly.Msg;
+
+        // builtin controls_repeat_ext
+        msg.CONTROLS_REPEAT_TITLE = lf("repeat %1 times");
+        msg.CONTROLS_REPEAT_INPUT_DO = lf("_repeat_do"); // "do" in English
```

Comment:
for now, do not use ids (e.g. ``_repeat_do``) in the localized as it requires more changes on our end to support. Instead, please file a bug to support those ids. Let's decouple those from this PR. So this shoud stay as ``lf("do")``